### ISSUE TITLE
Missing documentation of `dispatch` function's stability 

### ISSUE THREAD

**Issue by forman** [reactjs/react.dev#6178]
First, I like the new React docs a lot! Unfortunately, I can no longer find the following information about `useReducer` that was once available in the docs:

_The `useReducer` hook returns an array with two elements: the current state and a dispatch function. The dispatch function is used to update the state by calling it with an action object. The dispatch function returned by `useReducer` has a lifecycle similar to that of the component it is used in. It is created when the component mounts and destroyed when the component unmounts. The dispatch function is guaranteed to be stable across re-renders._

Is this no longer the case?

On page [**Lifecycle of Reactive Effects**](https://react.dev/learn/lifecycle-of-reactive-effects) there is a note box saying _"In some cases, React knows that a value never changes..."_ and mentioning `useState` and `useRef`. It would be ideal if the documentation would explicitly enumerate all the stable return values from hooks, i.e., the ones that have the same lifecycle their component.

**Comment by rickhanlonii** [comment]
Yeah, may be worth mention in the caveats of the dispatch function docs @Afrin127329 mentioned here: https://react.dev/reference/react/useReducer#dispatch. Want to submit a PR?
